src/libopensc/simpletlv.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * simpletlv.c: Simple TLV encoding and decoding functions
+ *
+ * Copyright (C) 2016  Red Hat, Inc., Robert Relyea <rrelyea@redhat.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <assert.h>
+#include <stdlib.h>
+
+#include "internal.h"
+#include "simpletlv.h"
+
+/*
+ * Put a tag/length record to a file in Simple TLV based on the  datalen
+ * content length.
+ */
+int
+sc_simpletlv_put_tag(u8 tag, size_t datalen, u8 *out, size_t outlen, u8 **ptr)
+{
+	u8 *p = out;
+
+	if (outlen < 2 || (outlen < 4 && datalen >= 0xff))
+		return SC_ERROR_INVALID_ARGUMENTS;
+
```

Comment:
That makes sense, according to the [specification](http://geek-mag.com/posts/251450/).